File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/CategoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface CategoryRow {
    CategoryId?: number;
    CategoryName?: string;
    CreateTime?: string;
    UpdateTime?: string;
}

export abstract class CategoryRow {
    static readonly idProperty = 'CategoryId';
    static readonly nameProperty = 'CategoryName';
    static readonly localTextPrefix = 'BookManage.Category';
    static readonly lookupKey = 'BookManage.Category';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CategoryRow>('BookManage.Category') }
    static async getLookupAsync() { return getLookupAsync<CategoryRow>('BookManage.Category') }

    static readonly deletePermission = 'BookManage:Category';
    static readonly insertPermission = 'BookManage:Category';
    static readonly readPermission = 'BookManage:Category';
    static readonly updatePermission = 'BookManage:Category';

    static readonly Fields = fieldsProxy<CategoryRow>();
}

File: LMIS/LMIS.Web/Modules/BookManage/Category/CategoryRow.cs
using LMIS.Modules.BookManage;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace LMIS.BookManage;

[ConnectionKey("Default"), Module("BookManage"), TableName("categories")]
[DisplayName("Category"), InstanceName("Category")]
[ReadPermission(BookManagePermissionKeys.Category)]
[ModifyPermission(BookManagePermissionKeys.Category)]
[ServiceLookupPermission("*")]
[LookupScript(Permission = "*")]
public sealed class CategoryRow : Row<CategoryRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Category Id"), Identity, IdProperty]
    public long? CategoryId { get => fields.CategoryId[this]; set => fields.CategoryId[this] = value; }

    [DisplayName("Category Name"), Size(50), NotNull, QuickSearch, NameProperty]
    public string CategoryName { get => fields.CategoryName[this]; set => fields.CategoryName[this] = value; }

    [DisplayName("Create Time"), NotNull]
    public DateTime? CreateTime { get => fields.CreateTime[this]; set => fields.CreateTime[this] = value; }

    [DisplayName("Update Time"), NotNull]
    public DateTime? UpdateTime { get => fields.UpdateTime[this]; set => fields.UpdateTime[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int64Field CategoryId;
        public StringField CategoryName;
        public DateTimeField CreateTime;
        public DateTimeField UpdateTime;

    }
}